lib/gui/components/modal/services/modal.js

```diff
@@ -55,23 +59,33 @@ module.exports = function($uibModal, $q) {
     return {
       close: modal.close,
       result: $q((resolve, reject) => {
-        modal.result
-          .then(resolve)
-          .catch((error) => {
+        modal.result.then((value) => {
+          AnalyticsService.logEvent('Modal accepted', value);
+
+          resolve(value);
+        }).catch((error) => {
+
+          // For some annoying reason, UI Bootstrap Modal rejects
+          // the result reason if the user clicks on the backdrop
+          // (e.g: the area surrounding the modal).
+          if (error === 'escape key press' || error === 'backdrop click') {
+            AnalyticsService.logEvent('Modal rejected', {
+              method: error
+            });
 
             // Bootstrap doesn't 'resolve' these but cancels the dialog;
-            // therefore call 'resolve' here applied to 'false'.
-            if (error === 'escape key press' || error === 'backdrop click') {
-              resolve();
+            // therefore call 'resolve' here.
+            resolve();
+
+          } else {
+            AnalyticsService.logEvent('Modal rejected', {
+              value: error
```

Comment:
?? Just a few lines previously you used `method: error`, but here you're using `value: error`?
Should they be made consistent, or could you add a code-comment explaining why they're not consistent? ;)  (assuming I'm not missing anything obvious)